File: 409_LongestPalindrome.cpp
class Solution {
public:
    int longestPalindrome(string s) {
        map<char, int> counter;
		for(auto c : s)
			counter[c]++;
		bool hasOdd(false);
		int res(0);
		for(auto &item : counter){
			if(item.second & 1){
				hasOdd = true;
				item.second--;
			}
			res += item.second;
		}
		return hasOdd ? 1 + res : res;
    }
};

int longestPalindrome(string s) {
    int odds = 0;
    for (char c='A'; c<='z'; c++)
        odds += count(s.begin(), s.end(), c) & 1;
    return s.size() - odds + (odds > 0);
}

int longestPalindrome(string s) {
    vector<int> count(256);
    for (char c : s)
        ++count[c];
    int odds = 0;
    for (int c : count)
        odds += c & 1;
    return s.size() - odds + (odds > 0);
}

int longestPalindrome(string s) {
    vector<int> count(256);
    int odds = 0;
    for (char c : s)
        odds += ++count[c] & 1 ? 1 : -1;
    return s.size() - odds + (odds > 0);
}

File: 409_LongestPalindrome.py
# 22.86%
import collections
class Solution(object):
    def longestPalindrome(self, s):
        """
        :type s: str
        :rtype: int
        """
        c = collections.Counter(s)
        res, odd = 0, False
        for _, num in c.items():
            if num % 2 != 0:
                odd = True
                num -= 1
            res += num
        return res if not odd else res + 1
		
def longestPalindrome(self, s):
    odds = sum(v & 1 for v in collections.Counter(s).values())
    return len(s) - odds + bool(odds)

def longestPalindrome(self, s):
    use = sum(v & ~1 for v in collections.Counter(s).values())
    return use + (use < len(s))

def longestPalindrome(self, s):
    counts = collections.Counter(s).values()
    return sum(v & ~1 for v in counts) + any(v & 1 for v in counts)